### PR TITLE
add queue_generator_op, dequeue_op, enqueue_op and ut

### PR DESCRIPTION
**Pull request by sandyhouse**
PR types: 'New features'

PR changes: 'Ops'

Describe: Add queue_generator_op, dequeue_op, enqueue_op, which are the basic of pipeline.
The queue is used to transfer tensor data between devices.
The underlying of a queue is a LoDTensorBlockingqueue.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_queue.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import numpy as np
+import os
+import unittest
+
+import paddle.fluid as fluid
+import paddle.fluid.layers as layers
+import paddle.fluid.core as core
+
+
+class TestQueue(unittest.TestCase):
```
Comment (gavin1332) [inline python/paddle/fluid/tests/unittests/test_queue.py:26]: split into `TestEnqueue`, `TestDequeue` and `TestQueueGenerator`, one operator one unittest.

Comment (gavin1332) [inline python/paddle/fluid/tests/unittests/test_queue.py:26]: unclear name of a TestCase

**Review by gavin1332 — APPROVED** [review]
LGTM

**Review by phlrain — APPROVED** [review]

**Review by Xreki — APPROVED** [review]
LGTM for the modification of framework.py

**Review by luotao1 — APPROVED** [review]

**Pull request merged**